integration_tests/testsuite/driver.py

```diff
@@ -73,15 +73,19 @@ def _main():
         logging.debug('Deploying app!')
         deploy_url = deploy_app(args.image, args.directory)
 
-    if deploy_url is not '' and not deploy_url.startswith('https://'):
+    if args.deploy and not deploy_url:
+        logging.info('Defaulting to provided URL parameter.')
+        deploy_url = test_util.get_default_url()
+
+    if deploy_url and not deploy_url.startswith('https://'):
         deploy_url = 'https://' + deploy_url
 
-    if args.url is not None and args.url is not '':
+    if args.url and args.url is not '':
```

Comment:
you can drop the later part of check args.url is not ''
 if args.url should return false for args.url = ''